src/korma_encrypted/core.clj

```diff
@@ -34,32 +34,54 @@
 (defn- encrypted-name [field]
   (keyword (str "encrypted_" (name field))))
 
-(defn- get-encrypted-data-encryption-key []
-  (-> (korma/select data-encryption-keys)
-      first
-      :data_encryption_key))
+(defn- get-encrypted-data-encryption-key [pk]
+  (:data_encryption_key (first (korma/select data-encryption-keys
+                               (korma/where {:pk pk})))))
 
-(def get-data-box
+(defn- most-recent-data-encryption-key []
+  (first (korma/select data-encryption-keys
+                       (korma/order :pk :desc))))
+
+(def get-data-encryption-box
   (memoize
-    (fn [key-service]
-      (let [data-encryption-key (decrypt key-service (get-encrypted-data-encryption-key))]
+    (fn [key-service pk]
+      (let [data-encryption-key (decrypt key-service (get-encrypted-data-encryption-key pk))]
         (SecretBox. (str->secretkey data-encryption-key))))))
 
-(defn prepare-values [field key-service values]
-  (let [box (get-data-box key-service)]
-    (if (contains? values field)
-      (let [unencrypted-value (field values)]
+(defn update-encrypted-entity [ent pk fields]
```

Comment:
We addressed this and added a test for it.  Now it will return 0 (the number of rows that were updated) the same as `update` does.